src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -151,6 +152,46 @@ bool implicitConversionExists(BoundExpression expression, TypeSymbol type)
                     return c.IsImplicit;
                 }
             }
+
+            private static BoundStatement ConstructThrowSwitchExpressionExceptionHelperCall(SyntheticBoundNodeFactory factory, BoundExpression unmatchedValue)
+            {
+                var module = factory.ModuleBuilderOpt!;
```

Comment:
We should use an assertion rather than suppression here. Also applies to added methods below
We try to use suppressions sparingly: 1. inside assertions, 2. if the non-nullness is implied by preceding logic/check in method but compiler is just unable to see it

Reply:
@jcouv This was a copy from:

https://github.com/dotnet/roslyn/blob/534c0e8d04e58044c66ef4f66eeafd255a18749d/src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs#L41

But I agree.